examples/md-flexible/src/ParticleCommunicator.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * @file ParticleCommunicator.h
+ * @author J. Körner
+ * @date 28.07.2021
+ */
+#include "ParticleCommunicator.h"
+
+#include <vector>
+
+#include "ParticleSerializationTools.h"
+
+ParticleCommunicator::ParticleCommunicator(const autopas::AutoPas_MPI_Comm &communicator)
+    : _communicator(communicator) {}
+
+void ParticleCommunicator::sendParticles(const std::vector<ParticleType> &particles, const int &receiver) {
+  std::vector<char> buffer;
+
+  for (auto &particle : particles) {
+    ParticleSerializationTools::serializeParticle(particle, buffer);
+  }
+
+  sendDataToNeighbour(buffer, receiver);
+}
+
+void ParticleCommunicator::receiveParticles(std::vector<ParticleType> &receivedParticles, const int &source) {
+  std::vector<char> receiveBuffer;
+
+  receiveDataFromNeighbour(source, receiveBuffer);
+
+  if (!receiveBuffer.empty()) {
+    ParticleSerializationTools::deserializeParticles(receiveBuffer, receivedParticles);
+  }
+}
+
+void ParticleCommunicator::waitForSendRequests() {
+  std::vector<autopas::AutoPas_MPI_Status> sendStates;
+  sendStates.resize(_sendRequests.size());
+  autopas::AutoPas_MPI_Waitall(_sendRequests.size(), _sendRequests.data(), sendStates.data());
+  _sendRequests.clear();
+  _sendBuffers.clear();
+}
+
+void ParticleCommunicator::sendDataToNeighbour(std::vector<char> sendBuffer, const int &neighbour) {
```

Comment:
is it on purpose that you take a copy of the `sendBuffer`? Wouldn't a reference make more sense? You seem to copy it to `_sendBuffers` anyways.
Also, you can make it `const`.

```suggestion
void ParticleCommunicator::sendDataToNeighbour(const std::vector<char> &sendBuffer, const int &neighbour) {
```